src/components/AddPeopleAndRoles/OrgPerson.vue

```diff
@@ -0,0 +1,533 @@
+  private addPersonOrgFormValid: boolean = true
+
+  // Address related properties
+  private inProgressMailingAddress:AddressIF
+  private inProgressDeliveryAddress:AddressIF
+  private inheritMailingAddress: boolean = true
+  private personAddressSchema: {} = addressSchema
+  private mailingAddressValid: boolean = false
+  private deliveryAddressValid: boolean = false
+  private reassignCompletingParty : boolean = false
+
+  // Roles
+  private isCompletingParty: boolean = false
+  private isIncorporator: boolean = false
+  private isDirector: boolean = false
+
+  readonly EntityTypes: {} = EntityTypes
+  readonly Roles: {} = Roles
+  readonly IncorporatorTypes: {} = IncorporatorTypes
+
+  // Rules
+  private readonly firstNameRules: Array<Function> = [
+    v => !!v || 'A first name is required',
+    v => !/^\s/g.test(v) || 'Invalid spaces', // leading spaces
+    v => !/\s$/g.test(v) || 'Invalid spaces' // trailing spaces
+  ]
+
+  private readonly middleNameRules: Array<Function> = [
+    v => !/^\s/g.test(v) || 'Invalid spaces', // leading spaces
+    v => !/\s$/g.test(v) || 'Invalid spaces' // trailing spaces
+  ]
+
+  private readonly lastNameRules: Array<Function> = [
+    v => !!v || 'A last name is required',
+    v => !/^\s/g.test(v) || 'Invalid spaces', // leading spaces
+    v => !/\s$/g.test(v) || 'Invalid spaces' // trailing spaces
+  ]
+
+  private readonly orgNameRules: Array<Function> = [
+    v => !!v || 'A firm name is required',
+    v => !/^\s/g.test(v) || 'Invalid spaces', // leading spaces
+    v => !/\s$/g.test(v) || 'Invalid spaces' // trailing spaces
+  ]
+
+  // Life cycle methods
+  private created (): void {
+    if (this.initialValue) {
+      this.orgPerson = { ...this.initialValue }
+      this.isDirector = this.orgPerson.roles.includes(Roles.DIRECTOR)
+      this.isIncorporator = this.orgPerson.roles.includes(Roles.INCORPORATOR)
+      this.isCompletingParty = this.orgPerson.roles.includes(Roles.COMPLETING_PARTY)
+      this.inProgressMailingAddress = { ...this.orgPerson.address.mailingAddress }
+      if (this.isDirector) {
+        this.inProgressDeliveryAddress = { ...this.orgPerson.address.deliveryAddress }
+        this.inheritMailingAddress = this.isSame(this.inProgressMailingAddress, this.inProgressDeliveryAddress)
+      }
+    }
+  }
+
+  // Event Handlers
+  private updateMailingAddress (val): void {
+    this.inProgressMailingAddress = val
+  }
+
+  private updateDeliveryAddress (val): void {
+    this.inProgressDeliveryAddress = val
+  }
+
+  private updateMailingAddressValidity (val: boolean): void {
+    this.mailingAddressValid = val
+  }
+
+  private updateDeliveryAddressValidity (val: boolean): void {
+    this.deliveryAddressValid = val
+  }
+
+  private assignCompletingPartyRole (): void {
+    if (this.isCompletingParty && this.existingCompletingParty &&
+      this.orgPerson.id !== this.existingCompletingParty.id) {
+      this.confirmReassignPerson()
+    }
+  }
+
+  // Methods
```

Comment:
Do you think there is value in expanding the comments on these methods?

Reply:
I wasn't confused by most of these methods and that's my litmus test for whether something needs comments, but then again helpful comments are always welcome.

Reply:
I think you guys did a great job naming methods here and adding comments wouldn't add huge value.

Reply:
ok great, good to hear! I guess thats the true test, if someone can jump into the code and understand with ease!